usdot-fhwa-stol/carma-platform
# Arc 78 trajectory follower integration

# PR Details
## Description

The Trajectory Follower Wrapper is being tested on a physical vehicle. 
related PRs:
https://github.com/usdot-fhwa-stol/autoware.auto/pull/26
https://github.com/usdot-fhwa-stol/carma-vehicle-calibration/pull/128

## Related GitHub Issue







## Related Jira Key

(https://usdot-carma.atlassian.net/browse/ARC-78)
## Motivation and Context

Integrate the Trajectory Follower controller of Autoware, that has a MPC based lateral controller, to improve vehicle's performance.

## How Has This Been Tested?
This controller is tested on Blue Lexus. 
Notes for testing:
- Ensure the calibration repo is on the correct branch, to receive configuration parameters
- In Subsystem Controller, replace "pure_pursuit_wraper" with "trajectory_follower_wrapper"
- Make sure to use the localization mode "GNSS only with NDT initialization" so /tf topic has correct information
- Check the log to see if MPC is being solved at each step.



## Types of changes




## Checklist:




- [] I have added tests to cover my changes.
